src/io/src/DataLoaderBridge.cpp

```diff
@@ -0,0 +1,274 @@
+    std::sort(depths.begin(), depths.end());
+    data_size = depths.size();
+    haveDepth = (data_size > 0);
+  }
+
+  this->isLoop = pt.get<bool>("option.isLoop", true);
+
+  if(!checkConsistency()){
+    outError("Provided data is not consistent, all must be a file or size of each kind of data must be equal!");
+    _newData = false;
+  }
+
+  this->frameRate = pt.get<double>("camera_info.frame_rate", -1);
+  this->cameraInfo.width = pt.get<int>("camera_info.width", 640);
+  this->cameraInfo.height = pt.get<int>("camera_info.height", 480);
+
+  this->cameraInfo.roi.width = this->cameraInfo.width;
+  this->cameraInfo.roi.height = this->cameraInfo.height;
+  this->cameraInfo.roi.x_offset = 0;
+  this->cameraInfo.roi.y_offset = 0;
+
+  std::string line = pt.get<std::string>("camera_info.matrix");
+  std::replace(line.begin(), line.end(), ',', ' ');
+
+  std::stringstream sstr(line);
+  std::vector<double> cameraMatrix{std::istream_iterator<double>(sstr),
+                                   std::istream_iterator<double>()};
+  std::copy(cameraMatrix.begin(), cameraMatrix.end(), this->cameraInfo.K.begin());
+
+  line = pt.get<std::string>("camera_info.distortion");
+  std::replace(line.begin(), line.end(), ',', ' ');
+  sstr.str(line);
+  std::vector<double> cameraDistortion{std::istream_iterator<double>(sstr),
+                                       std::istream_iterator<double>()};
+  std::copy(cameraDistortion.begin(), cameraDistortion.end(), this->cameraInfo.D.begin());
+
+  this->depth_scaling_factor = pt.get<int>("camera_info.depth_scaling_factor", 3000);
+}
+
+bool DataLoaderBridge::setData(uima::CAS &tcas, uint64_t ts){
+  if(!newData())
+    return false;
+
+  outInfo("setData");
+
+  rs::SceneCas cas(tcas);
+
+  if(haveCloud){
+    pcl::PointCloud<pcl::PointXYZRGBA>::Ptr cloud_ptr (new pcl::PointCloud<pcl::PointXYZRGBA>);
+
+    std::string path;
+    if(!isCloudFile){
+      path = clouds[iterator];
+    }
+    else{
+      path = path_to_cloud;
+    }
+
+    if(pcl::io::loadPCDFile (path, *cloud_ptr) == -1){
+      outError("Could not load point cloud file as PCD type. Check path again!");
+    }
+
+    cas.set(VIEW_CLOUD, *cloud_ptr);
+  }
+
+  auto imageSize = cv::Size(this->cameraInfo.width, this->cameraInfo.height);
+
+  if(haveRGB){
+    std::string path;
+    if(!isRGBFile){
+      path = images[iterator];
+    }
+    else{
+      path = path_to_rgb;
+    }
+
+    this->color = cv::imread(path);
+    cv::resize(color, color, imageSize, 0, 0, cv::INTER_NEAREST);
+    cas.set(VIEW_COLOR_IMAGE, color);
+  }
+
+  if(haveDepth){
+    std::string path;
+    if(!isDepthFile){
+      path = depths[iterator];
+    }
+    else{
+      path = path_to_depth;
+    }
+
+    this->depth = cv::imread(path,  CV_LOAD_IMAGE_ANYDEPTH);
+    cv::resize(depth, depth, imageSize, 0, 0, cv::INTER_NEAREST);
+
+    if (depth.type() == CV_16UC1)
+      depth.convertTo(depth, CV_16UC1, this->depth_scaling_factor / 65535, 0);
```

Comment:
We probably don't need it for 16bit images, but 255 values is too few for whole depth range, so for consistency between image data types I decided to leave it for all other types.
@anindex, could you change this, since you have a direct access to this pr, to

`depth.convertTo(depth, -1, this->depth_scaling_factor, 0);`

For all `depth.type()`? And change default value in readConfig to 1, as well as a value in ini file